SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -2046,6 +2046,9 @@ void CSolver::AdaptCFLNumber(CGeometry **geometry,
   const su2double CFLMax            = config->GetCFL_AdaptParam(3);
   const bool fullComms              = (config->GetComm_Level() == COMM_FULL);
 
+  /* Number of iterations considered to check for stagnation. */
+  const auto Res_Count = min(100ul, config->GetnInner_Iter()-1);
```

Comment:
I added this because for unsteady and multiphysics the inner iterations may not get to 100